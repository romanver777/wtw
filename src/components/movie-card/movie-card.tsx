import React, { ReactElement } from "react";

import { PAGE } from "../../helpers/const";
import { FilmType } from "../../types/types";
import Header from "../header/header";
import Logo from "../logo/logo";
import UserBlock from "../user-block/user-block";

const MovieCard: React.FC<FilmType> = ({ film }): ReactElement => {
  return (
    <section className="movie-card">
      <div className="movie-card__bg">
        <img src={film.posterUrl} alt={film.nameRu} />
      </div>

      <h1 className="visually-hidden">WTW</h1>

      <Header page={PAGE.MOVIE_CARD}>
        <Logo clName="" />
        <UserBlock />
      </Header>

      <div className="movie-card__wrap">
        <div className="movie-card__info">
          <div className="movie-card__poster">
            <img
              src={film.posterUrl}
              alt={film.nameRu}
              width="218"
              height="327"
            />
          </div>

          <div className="movie-card__desc">
            <h2 className="movie-card__title">{film.nameRu}</h2>
            <p className="movie-card__meta">
              {film.genres.map((item, ind) => {
                return (
                  <span className="movie-card__genre" key={ind}>
                    {item.genre}
                  </span>
                );
              })}
              <span className="movie-card__year">{film.year}</span>
            </p>

            <div className="movie-card__buttons">
              <button
                className="btn btn--play movie-card__button"
                type="button"
              >
                <svg viewBox="0 0 19 19" width="19" height="19">
                  <use xlinkHref="#play-s"></use>
                </svg>
                <span>Play</span>
              </button>
              <button
                className="btn btn--list movie-card__button"
                type="button"
              >
                <svg viewBox="0 0 19 20" width="19" height="20">
                  <use xlinkHref="#add"></use>
                </svg>
                <span>My list</span>
              </button>
            </div>
          </div>
        </div>
      </div>
    </section>
  );
};

export default MovieCard;
