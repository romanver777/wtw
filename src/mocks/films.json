{
  "films": [
    {
      "filmId": 1302121,
      "nameRu": "Соединённые Штаты против Билли Холидей",
      "nameEn": "The United States vs. Billie Holiday",
      "year": "2021",
      "filmLength": "2:10",
      "countries": [
        {
          "country": "США"
        }
      ],
      "genres": [
        {
          "genre": "драма"
        },
        {
          "genre": "биография"
        },
        {
          "genre": "музыка"
        }
      ],
      "rating": "5.8",
      "ratingVoteCount": 586,
      "posterUrl": "https://kinopoiskapiunofficial.tech/images/posters/kp/1302121.jpg",
      "posterUrlPreview": "https://kinopoiskapiunofficial.tech/images/posters/kp_small/1302121.jpg",
      "ratingChange": null
    },
    {
      "filmId": 1171895,
      "nameRu": "Неуязвимый (сериал)",
      "nameEn": "Invincible",
      "year": "2021-...",
      "filmLength": null,
      "countries": [
        {
          "country": "США"
        }
      ],
      "genres": [
        {
          "genre": "мультфильм"
        },
        {
          "genre": "ужасы"
        },
        {
          "genre": "фантастика"
        }
      ],
      "rating": "8.1",
      "ratingVoteCount": 1549,
      "posterUrl": "https://kinopoiskapiunofficial.tech/images/posters/kp/1171895.jpg",
      "posterUrlPreview": "https://kinopoiskapiunofficial.tech/images/posters/kp_small/1171895.jpg",
      "ratingChange": null
    },
    {
      "filmId": 1245486,
      "nameRu": "Невероятные (сериал)",
      "nameEn": "The Nevers",
      "year": "2021-...",
      "filmLength": null,
      "countries": [
        {
          "country": "Великобритания"
        },
        {
          "country": "США"
        }
      ],
      "genres": [
        {
          "genre": "фантастика"
        },
        {
          "genre": "фэнтези"
        },
        {
          "genre": "боевик"
        }
      ],
      "rating": "7.0",
      "ratingVoteCount": 882,
      "posterUrl": "https://kinopoiskapiunofficial.tech/images/posters/kp/1245486.jpg",
      "posterUrlPreview": "https://kinopoiskapiunofficial.tech/images/posters/kp_small/1245486.jpg",
      "ratingChange": null
    },
    {
      "filmId": 1015471,
      "nameRu": "Райя и последний дракон",
      "nameEn": "Raya and the Last Dragon",
      "year": "2021",
      "filmLength": "1:47",
      "countries": [
        {
          "country": "США"
        }
      ],
      "genres": [
        {
          "genre": "мультфильм"
        },
        {
          "genre": "фэнтези"
        },
        {
          "genre": "боевик"
        }
      ],
      "rating": "7.6",
      "ratingVoteCount": 43386,
      "posterUrl": "https://kinopoiskapiunofficial.tech/images/posters/kp/1015471.jpg",
      "posterUrlPreview": "https://kinopoiskapiunofficial.tech/images/posters/kp_small/1015471.jpg",
      "ratingChange": null
    },
    {
      "filmId": 571335,
      "nameRu": "Бесстыжие (сериал)",
      "nameEn": "Shameless",
      "year": "2011-2021",
      "filmLength": "0:55",
      "countries": [
        {
          "country": "США"
        }
      ],
      "genres": [
        {
          "genre": "драма"
        },
        {
          "genre": "комедия"
        }
      ],
      "rating": "8.6",
      "ratingVoteCount": 207954,
      "posterUrl": "https://kinopoiskapiunofficial.tech/images/posters/kp/571335.jpg",
      "posterUrlPreview": "https://kinopoiskapiunofficial.tech/images/posters/kp_small/571335.jpg",
      "ratingChange": null
    },
    {
      "filmId": 1328036,
      "nameRu": "Happy End (сериал)",
      "nameEn": null,
      "year": "2021-...",
      "filmLength": "0:35",
      "countries": [
        {
          "country": "Россия"
        }
      ],
      "genres": [
        {
          "genre": "драма"
        },
        {
          "genre": "комедия"
        }
      ],
      "rating": "7.4",
      "ratingVoteCount": 3282,
      "posterUrl": "https://kinopoiskapiunofficial.tech/images/posters/kp/1328036.jpg",
      "posterUrlPreview": "https://kinopoiskapiunofficial.tech/images/posters/kp_small/1328036.jpg",
      "ratingChange": null
    },
    {
      "filmId": 1045583,
      "nameRu": "Конёк-Горбунок",
      "nameEn": null,
      "year": "2021",
      "filmLength": "1:51",
      "countries": [
        {
          "country": "Россия"
        }
      ],
      "genres": [
        {
          "genre": "фэнтези"
        },
        {
          "genre": "комедия"
        },
        {
          "genre": "приключения"
        }
      ],
      "rating": "6.8",
      "ratingVoteCount": 100901,
      "posterUrl": "https://kinopoiskapiunofficial.tech/images/posters/kp/1045583.jpg",
      "posterUrlPreview": "https://kinopoiskapiunofficial.tech/images/posters/kp_small/1045583.jpg",
      "ratingChange": null
    },
    {
      "filmId": 1274271,
      "nameRu": "Родные",
      "nameEn": null,
      "year": "2020",
      "filmLength": "1:40",
      "countries": [
        {
          "country": "Россия"
        }
      ],
      "genres": [
        {
          "genre": "комедия"
        },
        {
          "genre": "драма"
        }
      ],
      "rating": "6.4",
      "ratingVoteCount": 20539,
      "posterUrl": "https://kinopoiskapiunofficial.tech/images/posters/kp/1274271.jpg",
      "posterUrlPreview": "https://kinopoiskapiunofficial.tech/images/posters/kp_small/1274271.jpg",
      "ratingChange": null
    },
    {
      "filmId": 767379,
      "nameRu": "Фарго (сериал)",
      "nameEn": "Fargo",
      "year": "2014-...",
      "filmLength": "0:53",
      "countries": [
        {
          "country": "США"
        }
      ],
      "genres": [
        {
          "genre": "триллер"
        },
        {
          "genre": "драма"
        },
        {
          "genre": "криминал"
        }
      ],
      "rating": "8.4",
      "ratingVoteCount": 119492,
      "posterUrl": "https://kinopoiskapiunofficial.tech/images/posters/kp/767379.jpg",
      "posterUrlPreview": "https://kinopoiskapiunofficial.tech/images/posters/kp_small/767379.jpg",
      "ratingChange": null
    },
    {
      "filmId": 1245524,
      "nameRu": "Зависнуть в Палм-Спрингс",
      "nameEn": "Palm Springs",
      "year": "2020",
      "filmLength": "1:30",
      "countries": [
        {
          "country": "США"
        },
        {
          "country": "Гонконг"
        }
      ],
      "genres": [
        {
          "genre": "комедия"
        },
        {
          "genre": "мелодрама"
        },
        {
          "genre": "фантастика"
        }
      ],
      "rating": "7.2",
      "ratingVoteCount": 130542,
      "posterUrl": "https://kinopoiskapiunofficial.tech/images/posters/kp/1245524.jpg",
      "posterUrlPreview": "https://kinopoiskapiunofficial.tech/images/posters/kp_small/1245524.jpg",
      "ratingChange": null
    },
    {
      "filmId": 4430018,
      "nameRu": "Абсолютное доказательство",
      "nameEn": "Absolute Proof",
      "year": "2021",
      "filmLength": "2:00",
      "countries": [
        {
          "country": "США"
        }
      ],
      "genres": [
        {
          "genre": "документальный"
        }
      ],
      "rating": null,
      "ratingVoteCount": null,
      "posterUrl": "https://kinopoiskapiunofficial.tech/images/posters/kp/4430018.jpg",
      "posterUrlPreview": "https://kinopoiskapiunofficial.tech/images/posters/kp_small/4430018.jpg",
      "ratingChange": null
    },
    {
      "filmId": 1437860,
      "nameRu": "Чикатило (сериал)",
      "nameEn": null,
      "year": "2020-2020",
      "filmLength": "0:50",
      "countries": [
        {
          "country": "Россия"
        }
      ],
      "genres": [
        {
          "genre": "триллер"
        },
        {
          "genre": "детектив"
        },
        {
          "genre": "драма"
        }
      ],
      "rating": "5.9",
      "ratingVoteCount": 7820,
      "posterUrl": "https://kinopoiskapiunofficial.tech/images/posters/kp/1437860.jpg",
      "posterUrlPreview": "https://kinopoiskapiunofficial.tech/images/posters/kp_small/1437860.jpg",
      "ratingChange": null
    },
    {
      "filmId": 1005008,
      "nameRu": "Майор Гром",
      "nameEn": null,
      "year": "2017",
      "filmLength": "0:25",
      "countries": [
        {
          "country": "Россия"
        }
      ],
      "genres": [
        {
          "genre": "короткометражка"
        },
        {
          "genre": "боевик"
        },
        {
          "genre": "криминал"
        }
      ],
      "rating": "7.0",
      "ratingVoteCount": 29251,
      "posterUrl": "https://kinopoiskapiunofficial.tech/images/posters/kp/1005008.jpg",
      "posterUrlPreview": "https://kinopoiskapiunofficial.tech/images/posters/kp_small/1005008.jpg",
      "ratingChange": null
    },
    {
      "filmId": 1291310,
      "nameRu": "В её глазах (мини-сериал)",
      "nameEn": "Behind Her Eyes",
      "year": "2021-2021",
      "filmLength": "0:50",
      "countries": [
        {
          "country": "Великобритания"
        }
      ],
      "genres": [
        {
          "genre": "триллер"
        },
        {
          "genre": "драма"
        },
        {
          "genre": "детектив"
        }
      ],
      "rating": "7.3",
      "ratingVoteCount": 12949,
      "posterUrl": "https://kinopoiskapiunofficial.tech/images/posters/kp/1291310.jpg",
      "posterUrlPreview": "https://kinopoiskapiunofficial.tech/images/posters/kp_small/1291310.jpg",
      "ratingChange": null
    },
    {
      "filmId": 1038789,
      "nameRu": "Охотник на монстров",
      "nameEn": "Monster Hunter",
      "year": "2020",
      "filmLength": "1:43",
      "countries": [
        {
          "country": "США"
        },
        {
          "country": "Китай"
        },
        {
          "country": "Германия"
        }
      ],
      "genres": [
        {
          "genre": "фэнтези"
        },
        {
          "genre": "боевик"
        },
        {
          "genre": "приключения"
        }
      ],
      "rating": "5.6",
      "ratingVoteCount": 24933,
      "posterUrl": "https://kinopoiskapiunofficial.tech/images/posters/kp/1038789.jpg",
      "posterUrlPreview": "https://kinopoiskapiunofficial.tech/images/posters/kp_small/1038789.jpg",
      "ratingChange": null
    },
    {
      "filmId": 1248786,
      "nameRu": "Аферистка",
      "nameEn": "I Care a Lot",
      "year": "2020",
      "filmLength": "1:58",
      "countries": [
        {
          "country": "США"
        },
        {
          "country": "Великобритания"
        }
      ],
      "genres": [
        {
          "genre": "триллер"
        },
        {
          "genre": "комедия"
        },
        {
          "genre": "криминал"
        }
      ],
      "rating": "6.6",
      "ratingVoteCount": 81569,
      "posterUrl": "https://kinopoiskapiunofficial.tech/images/posters/kp/1248786.jpg",
      "posterUrlPreview": "https://kinopoiskapiunofficial.tech/images/posters/kp_small/1248786.jpg",
      "ratingChange": null
    },
    {
      "filmId": 1193013,
      "nameRu": "Грань времени",
      "nameEn": "Synchronic",
      "year": "2019",
      "filmLength": "1:42",
      "countries": [
        {
          "country": "США"
        }
      ],
      "genres": [
        {
          "genre": "фантастика"
        },
        {
          "genre": "драма"
        }
      ],
      "rating": "6.2",
      "ratingVoteCount": 10361,
      "posterUrl": "https://kinopoiskapiunofficial.tech/images/posters/kp/1193013.jpg",
      "posterUrlPreview": "https://kinopoiskapiunofficial.tech/images/posters/kp_small/1193013.jpg",
      "ratingChange": null
    },
    {
      "filmId": 958500,
      "nameRu": "Охотник за разумом (сериал)",
      "nameEn": "Mindhunter",
      "year": "2017-2019",
      "filmLength": "1:00",
      "countries": [
        {
          "country": "США"
        }
      ],
      "genres": [
        {
          "genre": "триллер"
        },
        {
          "genre": "драма"
        },
        {
          "genre": "криминал"
        }
      ],
      "rating": "8.1",
      "ratingVoteCount": 50186,
      "posterUrl": "https://kinopoiskapiunofficial.tech/images/posters/kp/958500.jpg",
      "posterUrlPreview": "https://kinopoiskapiunofficial.tech/images/posters/kp_small/958500.jpg",
      "ratingChange": null
    },
    {
      "filmId": 1347194,
      "nameRu": "Звездный рубеж",
      "nameEn": "Cosmic Sin",
      "year": "2021",
      "filmLength": "1:28",
      "countries": [
        {
          "country": "США"
        }
      ],
      "genres": [
        {
          "genre": "фантастика"
        },
        {
          "genre": "боевик"
        }
      ],
      "rating": "3.4",
      "ratingVoteCount": 524,
      "posterUrl": "https://kinopoiskapiunofficial.tech/images/posters/kp/1347194.jpg",
      "posterUrlPreview": "https://kinopoiskapiunofficial.tech/images/posters/kp_small/1347194.jpg",
      "ratingChange": null
    },
    {
      "filmId": 1236082,
      "nameRu": "Ассасин: Битва миров",
      "nameEn": "Ci sha xiao shuo jia",
      "year": "2021",
      "filmLength": "2:10",
      "countries": [
        {
          "country": "Китай"
        }
      ],
      "genres": [
        {
          "genre": "фэнтези"
        },
        {
          "genre": "боевик"
        }
      ],
      "rating": "6.0",
      "ratingVoteCount": 645,
      "posterUrl": "https://kinopoiskapiunofficial.tech/images/posters/kp/1236082.jpg",
      "posterUrlPreview": "https://kinopoiskapiunofficial.tech/images/posters/kp_small/1236082.jpg",
      "ratingChange": null
    }
  ]
}
