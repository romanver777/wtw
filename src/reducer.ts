import movies from "./mocks/films.json";
import { FilmsType } from "./types/types";
import { GENRES } from "./helpers/const";

interface InitStateType {
  films: FilmsType;
  genre: string;
}

interface ActionType {
  SET_GENRE: string;
  GET_FILMS_BY_GENRE: string;
}

const InitialState: InitStateType = {
  films: [...movies],
  genre: GENRES[0],
};

const ActionType = {
  SET_GENRE: `SET_GENRE`,
  GET_FILMS_BY_GENRE: `GET_FILMS_BY_GENRE`,
};

const ActionCreator = {
  setGenre: (genre: string) => ({
    type: ActionType.SET_GENRE,
    payload: genre,
  }),
  getFIlmsByGenre: (genre, films) => ({
    type: ActionCreator.GET_FILMS_BY_GENRE,
    payload: [],
  }),
};

const reducer = (state = InitialState, action): InitStateType => {
  switch (action.type) {
    case ActionType.SET_GENRE:
      return {
        ...state,
        genre: action.payload,
      };
    default:
      return state;
  }
};

const Operation = {};

export { reducer, ActionCreator, Operation };
