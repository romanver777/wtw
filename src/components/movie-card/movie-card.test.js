/* eslint-disable @typescript-eslint/no-unsafe-call */
/* eslint-disable @typescript-eslint/no-unsafe-member-access */
/* eslint-disable @typescript-eslint/no-unsafe-assignment */
import React from "react";
import TestRenderer from "react-test-renderer";
import { createStore } from "redux";
import { Provider } from "react-redux";
import { reducer } from "../../reducer";
import { Router } from "react-router";
import history from "../../history";

import { MovieCard } from "./movie-card";
import films from "../../mocks/films.json";

const initState = {
  films,
  awaitFilm: null,
  currentFilm: films[0],
  hoveredFilm: null,
  activeGenre: "",
  genresList: null,
  tk: null,
  reviews: null,
  staff: null,
  hoveredVideoData: null,
};

const store = createStore(reducer, initState);

describe("MovieCard component", () => {
  const props = { isAuth: false, film: films[0] };
  const Trenderer = TestRenderer.create(
    <Provider store={store}>
      <Router history={history}>
        <MovieCard {...props} />
      </Router>
    </Provider>
  );
  it("renders correctly", () => {
    expect(Trenderer.toJSON()).toMatchSnapshot();
  });
});
