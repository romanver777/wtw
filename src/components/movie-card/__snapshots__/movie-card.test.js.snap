// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MovieCard component renders correctly 1`] = `
<section
  className="movie-card"
>
  <div
    className="movie-card__bg"
  >
    <img
      alt="Соединённые Штаты против Билли Холидей"
      src="https://kinopoiskapiunofficial.tech/images/posters/kp/1302121.jpg"
    />
  </div>
  <h1
    className="visually-hidden"
  >
    WTW
  </h1>
  <header
    className="page-header movie-card__head"
  >
    <div
      className="logo"
    >
      <a
        className="logo__link "
        href="#/"
        onClick={[Function]}
      >
        <span
          className="logo__letter logo__letter--1"
        >
          W
        </span>
        <span
          className="logo__letter logo__letter--2"
        >
          T
        </span>
        <span
          className="logo__letter logo__letter--3"
        >
          W
        </span>
      </a>
    </div>
    <div
      className="user-block"
    >
      <a
        className="user-block__link"
        href="#/login"
        onClick={[Function]}
      >
        Вход
      </a>
    </div>
  </header>
  <div
    className="movie-card__wrap"
  >
    <div
      className="movie-card__info"
    >
      <div
        className="movie-card__poster"
      >
        <img
          alt="Соединённые Штаты против Билли Холидей"
          height="327"
          src="https://kinopoiskapiunofficial.tech/images/posters/kp/1302121.jpg"
          width="218"
        />
      </div>
      <div
        className="movie-card__desc"
      >
        <h2
          className="movie-card__title"
        >
          Соединённые Штаты против Билли Холидей
        </h2>
        <p
          className="movie-card__meta"
        >
          <span
            className="movie-card__genre"
          >
            драма
          </span>
          <span
            className="movie-card__genre"
          >
            биография
          </span>
          <span
            className="movie-card__genre"
          >
            музыка
          </span>
          <span
            className="movie-card__year"
          >
            2021
          </span>
        </p>
        <div
          className="movie-card__buttons"
        >
          <button
            className="btn btn--play movie-card__button"
            onClick={[Function]}
            type="button"
          >
            <svg
              height="20"
              viewBox="0 0 19 20"
              width="19"
            >
              <use
                xlinkHref="#play-s"
              />
            </svg>
            <span>
              Смотреть
            </span>
          </button>
          <button
            className="btn btn--list movie-card__button"
            onClick={[Function]}
            type="button"
          >
            <svg
              height="20"
              viewBox="0 0 19 20"
              width="19"
            >
              <use
                xlinkHref="#add"
              />
            </svg>
            <span>
              Сохранить
            </span>
          </button>
        </div>
      </div>
    </div>
  </div>
</section>
`;
